docs/DeveloperGuide.md

```diff
@@ -1,38 +1,578 @@
+
+### 3.3. User component
+
+The class diagram below shows how User-related classes interact with each other. `User` object contains `name`, `age`, `contactNumber` attributes. Users are stored in UserList, which will be loaded and written on the file by `UserStorage` (inherits from `Storage`) whenever Upcycle runs or exits.
+All user-related commands operate mainly on a list of users (userList:UserList).
+
+![UserClassDiagram](images/UserClassDiagram.png)
+
+### 3.4. Item component
+
+...To be updated
+
+### 3.5. Transaction component
+
+The Class diagram below show how Transaction-related classes interact with each other. `Transaction` object contains `transactionId`, `itemName`, `itemId`, `borrower`, `duration`, `createdAt`, `returnedAt` 
+attributes. Among those, `transactionId` is created by `IdGenerator`'s static method and dates are parsed by `DateParser`, therefore, Transaction class depends on those two classes.
+Transactions are stored in `TransactionList`, which will be loaded and written on the file by `TransactionStorage` (inherits from `Storage`) whenever Upcycle runs or exits. All transaction-related 
+commands operate mainly on a list of transaction (transactionList:TransactionList)
+
+![TransactionClassDiagram](images/TransactionClassDiagram.png)
+
+### 3.6. Ui component
+
+The UI component handles anything related to user interaction.
+Primarily, it consists of two main operations: reading user inputs and printing responses to the user.
+For reading inputs, there is simply `UI::readInput`
+For printing responses, there are `UI::printResponse` and `UI::printErrorResponse`
+Further methods for printing specific responses that correspond to a user's behavior are defined for convenience.
+
+The design of the UI component is to simply act as a helper class i.e. one consists mainly of static methods that is called directly from the class.
+This makes sense as the UI class should not to be instantiated and is not inherited by other classes.
+
+Below, we detail the design of the UI class with a class diagram
+![UiDesign](images/UiDesign.png)
+
+
+### 3.7. Storage component
+
+Upcycle has three separate Storage class, dedicated for three types of object: ```UserStorage```, ```ItemStorage```, and ```TransactionStorage```. All of these inherit from an abstract class called ```Storage```. 
+The following  diagrams show more details about Storage classes: 
+
+![StorageClassDiagram](images/StorageClassDiagram.png)
+
+Upcycle stores the user's data, including the user list, item list, transaction list in three files ```user.txt```, ```item.txt```, ```transaction.txt```, respectively.
+The data will be loaded when running the program, and ONLY be written to the files when exiting the program correctly, otherwise, the data during the execution will be gone.
+These find can be found in ```data``` folder in the same directory with the folder containing project root.
+
```

Comment:
On the website: https://ay2223s1-cs2113-w12-2.github.io/tp/DeveloperGuide.html
there is an empty subsection! 

<img width="343" alt="image" src="https://user-images.githubusercontent.com/31297758/198495329-11b36c30-2db4-4971-be51-4410ab171277.png">